### PR TITLE
feat(RHIF-255): Add Zero State to image builder

### PR DESCRIPTION
**Pull request by adonispuente**
This PR adds zero state to image builder. 
To test use an account with zero systems, or change the logic for what the state is supposed to show and verify things look correct. 

This zero state should pop up when there are 0 images on the account.

Fixes https://issues.redhat.com/browse/HMS-2184

### CODE REVIEW

**Comment by croissanne** [comment]
Thank you! A few thoughts:

1. Now that edge is also integrating, I think we should figure out how those two are supposed to play together.
2. I wonder if we should show it when linking directly to the wizard
3. The beta prompt/popup is no longer visible if there's no images

File: src/Router.js [hunk 1]
```diff
@@ -11,28 +15,80 @@ const LandingPage = lazy(() => import('./Components/LandingPage/LandingPage'));
...
+    }
+  }, [hasSystems]);
   const edgeParityFlag = useFlag('edgeParity.image-list');
   return (
-    <Routes>
-      <Route path="*" element={<LandingPage />}>
-        <Route path="imagewizard/:composeId?" element={<CreateImageWizard />} />
-        <Route path="share/:composeId" element={<ShareImageModal />} />
-      </Route>
+    <Suspense
```
Comment (lucasgarfield) [inline src/Router.js:35]: I don't think that I want to do this at the <Router>... maybe we can move this to <LandingPage> instead?

**Comment by lucasgarfield** [comment]
Hi @adonispuente, would you mind squashing the second commit into the first one?

**Comment by lavocatt** [comment]
It would be nice to wait up until https://github.com/RedHatInsights/image-builder-frontend/pull/1320 is merged so that we can use the new hook to know how many images are in the images table.

File: src/Router.js [hunk 2]
```diff
@@ -1,17 +1,102 @@
...
+  }
+  const composes = data.data;
+  switch (from) {
+    case 'ZeroState':
+      // navigating to the LandinPage is only allowed from the ZeroState page
+      if (composes.length > ZERO_IMAGES) {
+        navigate(resolveRelPath(''));
+      }
+      break;
+    case 'LandinPage':
```
Comment (lucasgarfield) [inline src/Router.js]: Small typo here and in a few of the comments, this should be LandingPage (with a g :wink:).

File: src/Router.js [hunk 3]
```diff
@@ -1,17 +1,102 @@
...
+ *
+ * caveat:
+ * This component takes care of displaying itself the CreateImageWizard. Meaning
+ * that the wizard is not being displayed using the navigation system. This is
+ * valid for-now™ [2023-09] because the wizard is being displayed in a modal.
+ * This would most likely have to get updated when the wizard will get changed
+ * to an InPage mode.
+ */
+const AppZeroState = () => {
+  const [wizardOpenned, setWizardOpenned] = useState(false);
```
Comment (lucasgarfield) [inline src/Router.js]: It isn't an official convention in our code base yet but I usually like to start boolean variable names with `is`, what do you think of `isWizardOpened`?

Comment (lavocatt) [reply src/Router.js]: Very good idea.

File: src/Components/ShareImageModal/RegionsSelect.tsx [hunk 4]
```diff
@@ -82,7 +82,7 @@ const RegionsSelect = ({
   });
 
   if (!isSuccess) {
-    return undefined;
```
Comment (lucasgarfield) [inline src/Components/ShareImageModal/RegionsSelect.tsx:85]: Returning `undefined` should actually be allowed now, after e5896d8d9c56713a0a3668e80f44547830812175. You might need to update your dependencies with an `npm ci` to get your tsserver to not complain about it.

File: src/Components/ShareImageModal/ShareImageModal.tsx [hunk 5]
```diff
@@ -32,7 +32,7 @@ const ShareToRegionsModal = () => {
 
   if (!composeId) {
     handleClose();
-    return undefined;
```
Comment (lucasgarfield) [inline src/Components/ShareImageModal/ShareImageModal.tsx:35]: See comment above, `npm ci` should allow you to return undefined.

File: src/Router.tsx [hunk 6]
```diff
@@ -59,43 +58,42 @@ const AppZeroState = () => {
   );
 };
 
-const ZERO_IMAGES = 0;
-
```
Comment (lucasgarfield) [inline src/Router.tsx]: As in previous commit, please change to `LandingPage` with small 'g'. :wink:

File: tsconfig.json [hunk 7]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "outDir": "./dist/",
     "noImplicitAny": true,
-    "module": "es6",
+    "module": "esnext",
```
Comment (lucasgarfield) [inline tsconfig.json:5]: Could this have some implications on browser compatibility? What doesn't work with `es6`?

File: src/Router.tsx [hunk 8]
```diff
@@ -59,43 +58,42 @@ const AppZeroState = () => {
...
+type ZeroStateNavigationType = {
+  from: 'ZeroState' | 'LandinPage';
```
[the remark this answers is not in the document]
Comment (lavocatt) [reply src/Router.tsx:62]: How man, I'm happy you get to see those typos :smile:  How about the `Landin'Page` (just kidding)

File: tsconfig.json (hunk 7 above)
Comment (lavocatt) [reply tsconfig.json:5]: I have no clue :shrug:. If you change it back, eslint's gonna rant about our current lazy imports not being possible. So I guess we were not compatible before ? But without knowing it.

File: src/Router.tsx [hunk 9]
```diff
@@ -59,43 +58,42 @@ const AppZeroState = () => {
...
+        break;
+      case 'LandinPage':
+        // navigating to the ZeroState page is only allowed from the LandinPage
+        if (composes.length === ZERO_IMAGES) {
+          navigate(resolveRelPath('zerostate'));
+        }
+        break;
+    }
   }
+  return <></>;
```
Comment (lucasgarfield) [inline src/Router.tsx:95]: We should be able to return `undefined` now.

File: src/Components/ShareImageModal/ShareImageModal.tsx (hunk 5 above)
Comment (lavocatt) [reply src/Components/ShareImageModal/ShareImageModal.tsx:35]: It's working for eslint, not in my nvim, it's weird.

Comment (lavocatt) [reply src/Components/ShareImageModal/ShareImageModal.tsx:35]: Right, I updated my coc modules, it's now fine.

File: src/Router.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (lucasgarfield) [reply src/Router.tsx:62]: :rofl: Landin'Page sounds great!

File: tsconfig.json [hunk 10]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "outDir": "./dist/",
     "noImplicitAny": true,
-    "module": "es6",
+    "module": "es2020",
```
Comment (lucasgarfield) [inline tsconfig.json:5]: I'd like to do just a bit more research into this, but my initial understanding is es2020 should be very close to es6 so this is probably fine. Let's add a comment here as to why (`es6 does not allow dynamic imports, which we use for React lazy() loading/code splitting`? These config files tend to become very opinionated in mysterious ways as time passes.

**Comment by lavocatt** [comment]
@jrusz this PR might break the IQE tests

**Comment by jrusz** [comment]
> @jrusz this PR might break the IQE tests

There are errors in the unit tests. The image didn't get built because of that so I can't test the change.

**Comment by jrusz** [comment]
Also that commit doesn't look quite right. Idk about the content, but the message is definitely from a different commit.

**Comment by lavocatt** [comment]
ah! I messed up a rebase it seems .. I went too fast!

**Comment by lavocatt** [comment]
The unit tests are fixed @jrusz all good for you test it.

**Comment by jrusz** [comment]
> The unit tests are fixed @jrusz all good for you test it.

Thanks! I'll get on it.

**Comment by jrusz** [comment]
Lol it actually passed :smile: ... Since I know already where to update the ouiaID I'm going to do that and then change it a little bit to make it more stable.

**Review by lavocatt — CHANGES_REQUESTED** [review]
Quoting @croissanne: 
> Theoretically there are 3 "empty" states:  an empty page state (returning user who's images expired), the zeroState, and the unentitled page.The unentitled page is of no concern to us because as soon as you have RHEL you're entitled. The empty page state will go away with blueprints (since those don't expire). Meaning after blueprints we can do the new zerostate.

There's no good ways to distinguish between the 3 states rn, so I'm blocking this work up until the blueprints wizard is there.

**Pull request closed without merging**